src/utilities/api.rest.ts

```diff
@@ -0,0 +1,109 @@
+import axios, {
+  AxiosError,
+  AxiosResponse,
+  Method,
+} from 'axios';
+import {getToken} from "../helper/auth";
+
+class Api {
+  private base_url = 'https://api.zid.sa/v1' as string;
```

Comment:
I suggest defining this base URL in a common place containing constant links.

Reply:
I didn't want to use .env or something like that cuz, I think it will be a vulnerability issue, where all the constants will be in a single please and if one was found, all of them will also be found.
+ this class will be the only place responsible for Zid API. so defining the URL in other file and call it here won't be that effective.